js/data/bucket.js

```diff
@@ -81,7 +81,7 @@ function Bucket(options) {
  * @private
  */
 Bucket.prototype.addFeatures = function() {
-    for (var i = 0; i < this.features.length; i++) {
+    for (var i = this.features.length - 1; i >= 0; i--) {
```

Comment:
Hmm. Native does not work this way. [It iterates forward.](https://github.com/mapbox/mapbox-gl-native/blob/d5fb535058d4d97567b08064e0203f45c2c8370f/src/mbgl/style/style_bucket_parameters.cpp#L8-L8)

If this is affecting rendering there must be some other difference between native and JS that is reversed.